src/pal/src/arch/i386/context2.S

```diff
@@ -113,31 +113,20 @@ LOCAL_LABEL(Done_Restore_CONTEXT_FLOATING_POINT):
     movdqu xmm7, [eax + CONTEXT_Xmm7]
 LOCAL_LABEL(Done_Restore_CONTEXT_EXTENDED_REGISTERS):
 
-    // Restore CONTROL register(s)
-    mov   ecx, [eax + CONTEXT_Eip]
-    mov   [esp], ecx
+    // Restore Stack
+    mov   esp, [eax + CONTEXT_Esp]
 
-    mov   ecx, [eax + CONTEXT_Esp]
-    push  ecx
-    mov   ecx, [eax + CONTEXT_Ebp]
-    push  ecx
-
-    pop   ebp
-    pop   esp
+    // Create a minimal frame
+    push  DWORD PTR [eax + CONTEXT_Eip]
+    push  DWORD PTR [eax + CONTEXT_Ebp]
 
     // Restore INTEGER register(s)
-    mov   ecx, [eax + CONTEXT_Edi]
-    push  ecx
-    mov   ecx, [eax + CONTEXT_Esi]
-    push  ecx
-    mov   ecx, [eax + CONTEXT_Edx]
-    push  ecx
-    mov   ecx, [eax + CONTEXT_Ecx]
-    push  ecx
-    mov   ecx, [eax + CONTEXT_Ebx]
-    push  ecx
-    mov   ecx, [eax + CONTEXT_Eax]
-    push  ecx
+    push  DWORD PTR [eax + CONTEXT_Edi]
```

Comment:
Can't we just load the registers from the context (doing the eax read as the last one) instead of pushing all of them and the popping?